---
title: "Report Arnes Group: scRNA"
subtitle: '`r format(Sys.Date(),format="%d-%b-%Y")`'
output:
  bookdown::html_document2:
    toc: true
    toc_float:
      collapsed: true
      smooth_scroll: true 
      print: false
    toc_depth: 4
    number_sections: true
    highlight: tango
    df_print: paged
    code_folding: "show"
    self_contained: true
    keep_md: false
    encoding: 'UTF-8'
    css: "assets/report.css"
    staff_web: "https://www.bric.ku.dk/core-facilities/bioinformatics/about-us/"

---

```{r,child="assets/_header-lab.Rmd"}
```

<!-- ----------------------- Do not edit above this ----------------------- -->

```{r,echo=FALSE,include=FALSE}
# CUSTOM VARIABLES

# custom ggplot theme
theme_report_h <- function (base_size=12,base_family=NULL,colour="grey60") {
  theme_bw(base_size=base_size,base_family=base_family) %+replace%
    theme(
      panel.border=element_blank(),
      panel.grid.minor=element_blank(),
      panel.grid.major.x=element_blank(),
      legend.position="top",
      legend.direction="horizontal",
      legend.justification="center",
      strip.background=element_blank(),
      axis.ticks.y=element_blank(),
      axis.ticks.x=element_line(colour=colour),
      plot.caption=element_text(hjust=0,colour=colour,size=10),
      plot.title=element_text(colour=colour),
      plot.subtitle=element_text(colour=colour)
    )
}

# custom ggplot theme
theme_report <- theme_report_v <- function (base_size=12,base_family=NULL,colour="grey60") {
  theme_bw(base_size=base_size,base_family=base_family) %+replace%
    theme(
      panel.border=element_blank(),
      panel.grid.minor=element_blank(),
      panel.grid.major.x=element_blank(),
      legend.position="right",
      legend.direction="vertical",
      legend.justification="center",
      strip.background=element_blank(),
      axis.ticks.y=element_blank(),
      axis.ticks.x=element_line(colour=colour),
      plot.caption=element_text(hjust=0,colour=colour,size=10),
      plot.title=element_text(colour=colour),
      plot.subtitle=element_text(colour=colour)
    )
}

# custom ggplot theme
theme_simple_h <- function (base_size=12,base_family=NULL,colour="grey60") {
  theme_bw(base_size=base_size,base_family=base_family) %+replace%
    theme(
      panel.border=element_blank(),
      panel.grid=element_blank(),
      legend.justification="center",
      legend.position="top",
      legend.direction="horizontal",
      strip.background=element_blank(),
      axis.ticks=element_blank(),
      axis.text=element_blank(),
      axis.title=element_blank(),
      plot.caption=element_text(hjust=0,colour=colour,size=10),
      plot.title=element_text(colour=colour),
      plot.subtitle=element_text(colour=colour)
    )
}

# custom ggplot theme
theme_simple_v <- function (base_size=12,base_family=NULL,colour="grey60") {
  theme_bw(base_size=base_size,base_family=base_family) %+replace%
    theme(
      panel.border=element_blank(),
      panel.grid=element_blank(),
      legend.justification="center",
      legend.position="right",
      legend.direction="vertical",
      strip.background=element_blank(),
      axis.ticks=element_blank(),
      axis.text=element_blank(),
      axis.title=element_blank(),
      plot.caption=element_text(hjust=0,colour=colour,size=10),
      plot.title=element_text(colour=colour),
      plot.subtitle=element_text(colour=colour)
    )
}

#colours
col_sll_green <- "#fdf1e7"
col_sll_blue <- "#0093BD"
col_sll_orange <- "#EF7C00"
col_sll_green_light <- "#fdf1e7"
col_sll_blue_light <- "#e5f4f8"
col_sll_orange_light <- "#fdf1e5"

# project variables
rep_core_id <- "scRNA_00"
rep_report_version <- "1.0"
rep_request <- "Jonathan Baldan"
rep_request_email <- "Jonathan.Baldan@vub.be"
rep_pi <- "Luis Arnes"
rep_pi_email <- "Luis.Arnes@bric.ku.dk"
rep_org <- "BRIC"
rep_core <- "Juliana Assis"
rep_core_email <- "juliana.geraldo@ku.dk"

"#B70404";
"#E9F2D1"
"#f6f9ec"
"#96b43f"
"#424949"
"#f4f6f6"
"#5d6d7e"
"#fbfcfc"
"#E0E0E0"
```


<br>

::: boxy
__core ID:__ `r rep_core_id`   
__Report Version:__ `r rep_report_version`  
__Request by:__ `r paste0(rep_request," (",rep_request_email,")")`  
__Principal Investigator:__ `r paste0(rep_pi," (",rep_pi_email,")")`   
__Organisation:__ `r rep_org`  
__core Staff:__ `r paste0(rep_core," (",rep_core_email,")")`  
:::

<br>

# Setup

All libraries/packages


```{r,echo = T, eval = T, include=F, width = "100%", message=FALSE, warning=FALSE}
## Paths
path <- '/Users/flb202/Documents/GitHub/scRNA_01/'

## LIBRARIES
# data handling
library(dplyr)
library(tidyr)
library(stringr)

# tables
library(kableExtra) # complete table
library(formattable) # table with conditional formatting
library(DT) # interactive table

# graphics
library(ggplot2) # static graphics

# interactive graphics
library(highcharter)
library(plotly)
library(ggiraph) # convert ggplot to interactive
library(dygraphs) # time series
library(networkD3) # network graph
library(leaflet) # interactive maps
library(crosstalk) # linking plots

# analysis
library("devtools")
library("bsselectR")
library("dplyr")
library("tidyr")
library("DECIPHER")
library("gplots")
library("gridExtra")
library("grid")
library("ggpubr")
library("reshape2")
library("reshape")
library("ggrepel")
library("ggh4x")
library("RColorBrewer")
library("tidyverse")
library("DESeq2") # Bioconductor
library('ggfortify')
library("AnnotationDbi")
library("FactoMineR")
library("factoextra")
library("PCAtools")
library("Rqc")
library("gt")
library("clusterProfiler")
library("rnaseqGene")
library("plotly")
library("biomaRt")
library("ggridges")
library("viridis")
library("hrbrthemes")
library("DAtest")
library("EnhancedVolcano")
library("clusterProfiler")
library("msigdbr")
library("org.Mm.eg.db")
library("magrittr")
library("fgsea")

```


# Background

The study consist of 21 Mice \ 


```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/KU/BRIC/Projects/Arnes_SingleCell/seurat/01_Plots/Experimental_Desing.png')
```

## Data   
There are 21 Mice and 7 samples 

* Type of data  
scRNA 

* Data location \
Lehmann

* Reference data used 
Mouse

# Explanation of the Report

The report is divided into two parts: **methods** and **results**. In the results section, it is also possible to find more explanations. It is possible to download all plots and tables by clicking copy/save.

# Methods
Raw base call (BCL) files generated by Illumina sequencers were demultiplexed into FASTQ files using 10X Genomics' Single-Cell Gene Expression Software Cell Ranger version 7.2.0 ("mkfastq" pipeline, default parameters) [Ref1](https://www.10xgenomics.com/support/software/cell-ranger/latest/miscellaneous/cr-citations).

The nf-core sc-rnaseq pipeline, version 2.4.1 [Ref2](https://github.com/nf-core/scrnaseq/blob/2.4.1/CITATIONS.md
), was selected to perform a best-practice analysis pipeline for processing 10x Genomics single-cell RNA-seq data. 10x Genomics provided the reference genome sequenced:  Mouse reference (mm10) - 2020-A [Ref3](https://www.10xgenomics.com/support/software/cloud-analysis/latest/miscellaneous/CA-frequently-asked-questions).

Filtered gene expression matrices were then analysed using Scarfweb [Ref4](https://doi.org/10.1038/s41467-022-32097-3) and [Ref5](https://nygen.io/). Individual analyses for each sample were performed for quality control. Poor-quality cells were filtered using the auto_filter_cells function using default values [Ref4](https://doi.org/10.1038/s41467-022-32097-3). In addition, all cells with >20% mitochondrial counts were removed. 



```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/01_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/02_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/03_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/04_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/05_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/06_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/07_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/08_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/09_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/10_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/11_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/12_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/13_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/14_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/15_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/PNG/16_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/01_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/02_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/03_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/04_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/05_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/06_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/07_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/08_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/09_Plot.png')
```

```{r, echo=FALSE, fig.width = 20.5, fig.height = 12.58}
knitr::include_graphics('/Users/flb202/Documents/GitHub/scRNA_01/ScarfWeb/Plots/Snap/snapshots/10_Plot.png')
```


<!-- --------------------- Do not edit this and below ---------------------- -->

</br>

```{r,child="assets/_footer-lab.Rmd"}
```

```{r,eval=FALSE,echo=FALSE}
# manually run this to render this document to HTML
rmarkdown::render("core-report.Rmd")
# then run this to convert HTML to PDF (if needed)
#pagedown::chrome_print("core-report.html",output="core-report.pdf")
```
